From 7edf3981b53ed92e10deb7cc32e6f21670f337a1 Mon Sep 17 00:00:00 2001
From: Brett Mastbergen <bmastbergen@ciq.com>
Date: Fri, 2 Jan 2026 09:20:56 -0500
Subject: [PATCH] Bluetooth: L2CAP: fix "bad unlock balance" in
 l2cap_disconnect_rsp

jira VULN-154996
cve CVE-2023-53297
commit-author Min Li <lm0963hack@gmail.com>
commit 25e97f7b1866e6b8503be349eeea44bb52d661ce

conn->chan_lock isn't acquired before l2cap_get_chan_by_scid,
if l2cap_get_chan_by_scid returns NULL, then 'bad unlock balance'
is triggered.

	Reported-by: syzbot+9519d6b5b79cf7787cf3@syzkaller.appspotmail.com
Link: https://lore.kernel.org/all/000000000000894f5f05f95e9f4d@google.com/
	Signed-off-by: Min Li <lm0963hack@gmail.com>
	Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
(cherry picked from commit 25e97f7b1866e6b8503be349eeea44bb52d661ce)
	Signed-off-by: Brett Mastbergen <bmastbergen@ciq.com>
---
 net/bluetooth/l2cap_core.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/bluetooth/l2cap_core.c b/net/bluetooth/l2cap_core.c
index cbd1d56e28b7..ba1f0898b6f7 100644
--- a/net/bluetooth/l2cap_core.c
+++ b/net/bluetooth/l2cap_core.c
@@ -4403,7 +4403,6 @@ static inline int l2cap_disconnect_rsp(struct l2cap_conn *conn,
 
 	chan = l2cap_get_chan_by_scid(conn, scid);
 	if (!chan) {
-		mutex_unlock(&conn->chan_lock);
 		return 0;
 	}
 
-- 
2.47.3

